File: app/src/main/java/quintype/com/templatecollectionwithrx/models/story/Default.java
package quintype.com.templatecollectionwithrx.models.story;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Default implements Parcelable {

@SerializedName("hero-image")
@Expose
private HeroImage heroImage;
@SerializedName("headline")
@Expose
private String headline;

/**
* 
* @return
* The heroImage
*/
public HeroImage getHeroImage() {
return heroImage;
}

/**
* 
* @param heroImage
* The hero-image
*/
public void setHeroImage(HeroImage heroImage) {
this.heroImage = heroImage;
}

/**
* 
* @return
* The headline
*/
public String getHeadline() {
return headline;
}

/**
* 
* @param headline
* The headline
*/
public void setHeadline(String headline) {
this.headline = headline;
}

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeParcelable(this.heroImage, flags);
        dest.writeString(this.headline);
    }

    public Default() {
    }

    protected Default(Parcel in) {
        this.heroImage = in.readParcelable(HeroImage.class.getClassLoader());
        this.headline = in.readString();
    }

    public static final Creator<Default> CREATOR = new Creator<Default>() {
        @Override
        public Default createFromParcel(Parcel source) {
            return new Default(source);
        }

        @Override
        public Default[] newArray(int size) {
            return new Default[size];
        }
    };
}

File: app/src/main/java/quintype/com/templatecollectionwithrx/models/story/Default.kt
package quintype.com.templatecollectionwithrx.models.story

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

/**
 * Created TemplateCollectionWithRx by rakshith on 9/4/18.
 */

class Default : Parcelable {

    /**
     *
     * @return
     * The heroImage
     */
    /**
     *
     * @param heroImage
     * The hero-image
     */
    @SerializedName("hero-image")
    @Expose
    var heroImage: HeroImage? = null
    /**
     *
     * @return
     * The headline
     */
    /**
     *
     * @param headline
     * The headline
     */
    @SerializedName("headline")
    @Expose
    var headline: String? = null

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeParcelable(this.heroImage, flags)
        dest.writeString(this.headline)
    }

    constructor() {}

    protected constructor(`in`: Parcel) {
        this.heroImage = `in`.readParcelable(HeroImage::class.java.classLoader)
        this.headline = `in`.readString()
    }

    companion object CREATOR : Parcelable.Creator<Default> {
        override fun createFromParcel(parcel: Parcel): Default {
            return Default(parcel)
        }

        override fun newArray(size: Int): Array<Default?> {
            return arrayOfNulls(size)
        }
    }

}